addon/query-cache.js

```diff
@@ -51,18 +50,23 @@ export default class QueryCache {
           'query-url'
         );
         let result = this._createResult(payload, { url, params, method, cacheKey }, array);
-
+        //Add result to reverseCache.
         if (cacheKey) {
-          this._addResultToCache(result, cacheKey);
+          this._addResultToReverseCache(result, cacheKey);
         }
         return result;
       });
     }
 
     if (reload || cachedValue === undefined) {
+      //Store the promise in the cache until it is fulfilled and
+      //retrun same promise for subsequent request.
+      if (cacheKey) {
+        this._queryCache[cacheKey] = loadPromise;
+      }
       return loadPromise;
     } else {
-      return resolve(cachedValue);
+      return cachedValue;
```

Comment:
Do you think it would be clearer to rename this to `cachedPromise` now that it's promises we're caching?

Reply:
I agree. `cachedPromise` makes more sense here now. I will update variable name.